src/tasks/submit/task.rs

```diff
@@ -230,44 +222,33 @@ impl SubmitTask {
             let ru_block_number = sim_result.block.block_number();
             let host_block_number = self.constants.rollup_block_to_host_block_num(ru_block_number);
 
+            // Don't submit empty blocks
+            if sim_result.block.is_empty() {
+                debug!(ru_block_number, "received empty block - skipping");
+                continue;
+            }
+
+            // Create a span for the entire block processing operation
             let span = debug_span!(
                 "SubmitTask::loop",
                 ru_block_number,
                 host_block_number,
                 block_tx_count = sim_result.block.tx_count(),
             );
-            let guard = span.enter();
-
+            let _guard = span.enter();
             debug!(ru_block_number, "submit channel received block");
 
-            // Don't submit empty blocks
-            if sim_result.block.is_empty() {
-                debug!(ru_block_number, "received empty block - skipping");
-                continue;
-            }
-
-            // drop guard before await
-            drop(guard);
-
-            let Ok(Some(prev_host)) = self
-                .provider()
-                .get_block_by_number(host_block_number.into())
-                .into_future()
-                .instrument(span.clone())
-                .await
+            let prev_host_number = host_block_number - 1;
```

Comment:
is this a drive-by or was it related to the panic?

Reply:
It's another bug, actually, just was part of the same PR. I think it's the reason why the builder isn't submitting blocks in dev-net right now. I will build and submit this into a new PR since the span fixes are addressed in #113.